### ISSUE TITLE
Handling of input stream error

### ISSUE THREAD

**Issue by ghost** [AlanQuatermain/aqtoolkit#6]
I am new to Objective C and your streaming XML parser, but I ran into problem when network error occurs. I wanted to know if this is correct:
In AQXMLParser.m at line 963:
    _internal->error = [input streamError];

Should this be:
    _internal->error = [[input streamError] retain];

When this error occurs, then the parser is released, I get a crash in `-dealloc` at the line:
    [_internal->error release];

with this error message:
    -[CFError release]: message sent to deallocated instance

**Comment by AlanQuatermain** [comment]
Yes, good catch— it should indeed be retained. I'll tweak it and push the changes. In fact, I think there might be some other similar changes in the Kobo app to move across.

**Comment by marcusramberg** [comment]
Funny. I just fixed this error in our app and was about to report it :)
